clash-ghc/tests/Clash/Tests/GHC/ClashFlags.hs

```diff
@@ -0,0 +1,407 @@
+  _opt_edalize_fuzz             <- H.forAll genFuzzMode
+  _opt_errorExtra_fuzz          <- H.forAll genFuzzMode
+  _opt_escapedIds_fuzz          <- H.forAll genFuzzMode
+  _opt_evaluatorFuelLimit_fuzz  <- H.forAll genFuzzMode
+  _opt_forceUndefined_fuzz      <- H.forAll genFuzzMode
+  _opt_hdlDir_fuzz              <- H.forAll genFuzzMode
+  _opt_hdlSyn_fuzz              <- H.forAll genFuzzMode
+  _opt_inlineConstantLimit_fuzz <- H.forAll genFuzzMode
+  _opt_inlineFunctionLimit_fuzz <- H.forAll genFuzzMode
+  _opt_inlineLimit_fuzz         <- H.forAll genFuzzMode
+  _opt_inlineWFCacheLimit_fuzz  <- H.forAll genFuzzMode
+  _opt_intWidth_fuzz            <- H.forAll genFuzzMode
+  _opt_lowerCaseBasicIds_fuzz   <- H.forAll genFuzzMode
+  _opt_newInlineStrat_fuzz      <- H.forAll genFuzzMode
+  _opt_primWarn_fuzz            <- H.forAll genFuzzMode
+  _opt_renderEnums_fuzz         <- H.forAll genFuzzMode
+  _opt_specLimit_fuzz           <- H.forAll genFuzzMode
+  _opt_ultra_fuzz               <- H.forAll genFuzzMode
+
+  _dbg_countTransformations_fuzz <- H.forAll genFuzzMode
+  _dbg_historyFile_fuzz          <- H.forAll genFuzzMode
+  _dbg_invariants_fuzz           <- H.forAll genFuzzMode
+  _dbg_transformationInfo_fuzz   <- H.forAll genFuzzMode
+  _dbg_transformations_fuzz      <- H.forAll genFuzzMode
+  _dbg_transformationsFrom_fuzz  <- H.forAll genFuzzMode
+  _dbg_transformationsLimit_fuzz <- H.forAll genFuzzMode
+
+  opts0 <- H.forAll $
+    genClashOpts
+      ( _opt_aggressiveXOpt_fuzz
+      , _opt_aggressiveXOptBB_fuzz
+      , _opt_cachehdl_fuzz
+      , _opt_checkIDir_fuzz
+      , _opt_clear_fuzz
+      , _opt_componentPrefix_fuzz
+      , _opt_edalize_fuzz
+      , _opt_errorExtra_fuzz
+      , _opt_escapedIds_fuzz
+      , _opt_evaluatorFuelLimit_fuzz
+      , _opt_forceUndefined_fuzz
+      , _opt_hdlDir_fuzz
+      , _opt_hdlSyn_fuzz
+      , _opt_inlineConstantLimit_fuzz
+      , _opt_inlineFunctionLimit_fuzz
+      , _opt_inlineLimit_fuzz
+      , _opt_inlineWFCacheLimit_fuzz
+      , _opt_intWidth_fuzz
+      , _opt_lowerCaseBasicIds_fuzz
+      , _opt_newInlineStrat_fuzz
+      , _opt_primWarn_fuzz
+      , _opt_renderEnums_fuzz
+      , _opt_specLimit_fuzz
+      , _opt_ultra_fuzz
+      , _dbg_countTransformations_fuzz
+      , _dbg_historyFile_fuzz
+      , _dbg_invariants_fuzz
+      , _dbg_transformationInfo_fuzz
+      , _dbg_transformations_fuzz
+      , _dbg_transformationsFrom_fuzz
+      , _dbg_transformationsLimit_fuzz )
+
+  let flags = renderClashFlags opts0
+  H.footnote ("flags: " <> show flags)
+
+  (leftovers, warns, opts1) <- H.evalIO (runParse flags)
+
+  [] H.=== leftovers
+  [] H.=== warns
+
+  H.diff opts0 (==) opts1
+
+testCases :: [TestTree]
+testCases =
+  case $(testGroupGenerator) of
+    TestGroup _ tree -> tree
+    _ -> error "Internal error"
+
+tests :: TestTree
+tests =
+  testGroup
+    "Clash.Tests.GHC.ClashFlags"
+    [ -- tasty-th doesn't account for Hedgehog tests
```

Comment:
> `-- tasty-th doesn't account for Hedgehog tests`

Even if you prefix with `prop_` instead of `hprop_`, as they do in the hedgehog docs?

Reply:
Yeah, it turns it into a `QuickCheck` test.